tests/unit_tests/test_game.py

```diff
@@ -951,16 +951,76 @@ async def test_game_results(game: Game, players):
     assert len(result_dict["teams"]) == 2
 
     for team in result_dict["teams"]:
-        assert team["outcome"] == (
-            "VICTORY" if team["player_ids"] == [host_id]
-            else "DEFEAT"
-        )
+        outcome, army = ("VICTORY", 0) if team["player_ids"] == [host_id] else ("DEFEAT", 1)
+        assert team["outcome"] == outcome
+        assert team["army_results"] == [
+            ArmyResult(
+                team["player_ids"][0],
+                army,
+                outcome,
+                [],
+            )
+        ]
     assert result_dict["game_id"] == game.id
     assert result_dict["map_id"] == game.map_id
     assert result_dict["featured_mod"] == "faf"
     assert result_dict["sim_mod_ids"] == []
 
 
+async def test_team_game_results(game: Game, game_add_players):
+    game.state = GameState.LOBBY
+    players = (*game_add_players(game, 2, 0), *game_add_players(game, 2, 2))
+
+    await game.launch()
+    for i, player in enumerate(players):
+        for _ in players:
+            if i == 0:  # This player was the last one standing
+                await game.add_result(player.id, i, "victory", 1)
+            else:
+                await game.add_result(player.id, i, "defeat", 1)
+
+    game_results = await game.resolve_game_results()
+    result_dict = game_results.to_dict()
+
+    for i, team in enumerate(result_dict["teams"]):
```

Comment:
If it's not too overly inconvenient I would prefer the asserts to be as "non-programatic" or as "hard coded" as possible. That just prevents the test from using a functions behavior to test its behavior. For instance this test would pass if `result_dict["teams"]` returned an empty list. I think it's perfectly OK to have a bit of duplicate code in unit tests.

Reply:
More like this?

```python
assert result_dict["teams"][0]["outcome"] == "UNKNOWN"
assert result_dict["teams"][0]team["army_results"] == [
    ArmyResult(1, 0, "VICTORY", []),
    ...
    ArmyResult(4, 3, "DEFEAT", [])
]
```

Reply:
Yea or even
```python3
assert result_dict["teams"][0] == {
    "outcome": "VICTORY",
    "army_results": [
        ...
    ]
}
```